src/foam/dao/ManyToManyRelationshipDAO.js

```diff
@@ -18,51 +18,38 @@
 foam.CLASS({
   package: 'foam.dao',
   name: 'ManyToManyRelationshipDAO',
-  extends: 'foam.dao.RelationshipDAO',
+  extends: 'foam.dao.ProxyDAO',
 
   implements: [ 'foam.mlang.Expressions' ],
 
   documentation: 'Adapts a DAO based on a *:* Relationship.',
 
   properties: [
     'junctionProperty',
-    'joinDAOKey',
+    'junctionDAOKey',
     'targetProperty',
     'junctionCls',
+    'obj',
     {
-      name: 'joinDAO',
+      name: 'junctionDAO',
       getter: function() {
-        return this.__context__[this.joinDAOKey];
-      }
-    },
-    {
-      name: 'predicate',
-      documentation: `ManyToMany filtered querys are always "backward" to
-        match inverse-name property and source object's id.`,
-      getter: function() {
-        return this.EQ(
-          this.junctionCls[foam.String.constantize(this.relationship.inverseName)],
-          this.obj.id);
+        return this.__context__[this.junctionDAOKey];
       }
     }
   ],
 
   methods: [
-    function put(obj) {
-      var self = this;
-      return self.__context__[foam.String.daoize(self.of.name)].put(obj)
-          .then(function(obj) {
-            return self.delegate.put(self.relationship.adaptTarget(self.obj, obj))
-          })
-    },
     function find(id) {
-      return this.joinDAO.find(id);
+      var self = this;
+      return self.junctionDAO.select(self.MAP(self.junctionProperty)).then(function(map) {
```

Comment:
It _can_ be really expensive, but it has the added benefit of being correct :).

I pushed a different version that does a .find() on the junction to see if a relation exists, then moves on to the delegate, should be faster.